lib/web/apiserver.go

```diff
@@ -2624,6 +2589,24 @@ func (h *Handler) deleteClusterLock(
 	return OK(), nil
 }
 
+// SessionController restricts creation of sessions based on
+// cluster session control configuration(locks, connection limits, etc).
+type SessionController interface {
+	AcquireSessionContext(ctx context.Context, sctx *SessionContext, login, localAddr, remoteAddr string) (context.Context, error)
```

Comment:
IMO getting and returning `context.Context` is confusing. Can you at least mention in the comment what is being returned? Child context created from the passed parent? What will happen when I cancel it? Will the whole thing cancel?